test/e2e/plugin_lifecycle_test.go

```diff
@@ -0,0 +1,155 @@
+		secret := setup.CreateSecret(test.Ctx, clusterName,
+			test.WithSecretType(greenhouseapis.SecretTypeKubeConfig),
+			test.WithSecretData(map[string][]byte{greenhouseapis.KubeConfigKey: remoteKubeConfig}))
+		cluster := &greenhousev1alpha1.Cluster{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      clusterName,
+				Namespace: setup.Namespace(),
+			},
+		}
+		test.EventuallyCreated(test.Ctx, test.K8sClient, cluster)
+
+		testPluginDefinition := fixtures.NginxPluginDefinition
+		testPluginDefinition.ObjectMeta.Namespace = setup.Namespace() //namespace override
+
+		testPlugin := &greenhousev1alpha1.Plugin{
+			TypeMeta: metav1.TypeMeta{
+				Kind:       "Plugin",
+				APIVersion: greenhousev1alpha1.GroupVersion.String(),
+			},
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "test-nginx-plugin",
+				Namespace: setup.Namespace(),
+			},
+			Spec: greenhousev1alpha1.PluginSpec{
+				PluginDefinition: "nginx-18.1.7",
+				ReleaseNamespace: setup.Namespace(),
+				ClusterName:      secret.Name,
+				OptionValues: []greenhousev1alpha1.PluginOptionValue{
+					{
+						Name:  "replicaCount",
+						Value: &apiextensionsv1.JSON{Raw: []byte("1")},
+					},
+				},
+			},
+			Status: greenhousev1alpha1.PluginStatus{},
+		}
+
+		pluginDefinitionList := &greenhousev1alpha1.PluginDefinitionList{}
+		pluginList := &greenhousev1alpha1.PluginList{}
+		deploymentList := &appsv1.DeploymentList{}
+		ctx := test.Ctx
+
+		//Creating plugin definition
+		err := test.K8sClient.Create(ctx, testPluginDefinition)
+		Expect(err).NotTo(HaveOccurred())
+		err = test.K8sClient.List(ctx, pluginDefinitionList)
+		Expect(err).NotTo(HaveOccurred())
+		Expect(len(pluginDefinitionList.Items)).To(BeEquivalentTo(1))
+
+		//Creating plugin
+		err = test.K8sClient.Create(ctx, testPlugin)
+		Expect(err).NotTo(HaveOccurred())
+		err = test.K8sClient.List(ctx, pluginList)
+		Expect(err).NotTo(HaveOccurred())
+		Expect(len(pluginList.Items)).To(BeEquivalentTo(1))
+
+		//Checking deployment
+		err = remoteClient.List(ctx, deploymentList, client.InNamespace(setup.Namespace()))
+		Expect(err).NotTo(HaveOccurred())
+		SetDefaultEventuallyTimeout(60 * time.Second)
+		Eventually(func(g Gomega) bool {
+			err = remoteClient.List(ctx, deploymentList, client.InNamespace(setup.Namespace()))
+			g.Expect(err).NotTo(HaveOccurred())
+			g.Expect(len(deploymentList.Items) > 0).To(BeTrue())
+			return true
+		}).Should(BeTrue())
+
+		//Checking the name of deployment
+		nginxDeploymentExists := false
+		for _, deployment := range deploymentList.Items {
+			if strings.Contains(deployment.Name, "nginx") {
+				nginxDeploymentExists = true
+				Expect(deployment.Spec.Replicas).To(BeEquivalentTo(ptr[int32](1)))
+				break
+			}
+		}
+		Expect(nginxDeploymentExists).To(BeTrue())
+
+		//Updating replicas
+		err = test.K8sClient.List(ctx, pluginList)
+		Expect(err).NotTo(HaveOccurred())
+		Expect(len(pluginList.Items)).To(BeEquivalentTo(1))
+		testPlugin = &pluginList.Items[0]
```

Comment:
It's possible to fetch the plugin by it's namespaced name.
```suggestion
                namespacedName := types.NamespacedName{Name: testPlugin.Name, Namespace: testPlugin.Namespace}
		err = test.K8sClient.Get(ctx, namespacedName, &testPlugin )
		Expect(err).NotTo(HaveOccurred())
```

Reply:
Done